src/Compilers/CSharp/Portable/Symbols/FileIdentifier.cs

```diff
@@ -3,39 +3,93 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
 using System.Text;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp;
 
-internal readonly struct FileIdentifier
+internal sealed class FileIdentifier
 {
+    private record FileIdentifierData(string? EncoderFallbackErrorMessage, string DisplayFilePath, ImmutableArray<byte> FilePathChecksumOpt);
```

Comment:
>record [](http://example.com/codeflow?start=12&length=6)

Unless I am mistaken, we are not using any of the code generated for this type other than properties. Please switch to a regular class. Also, for private types like this, we often stick with public readonly fields (not even bothering with properties).

Reply:
Are primary ctors ok to use?

Reply:
> Are primary ctors ok to use?

As far as I know.